app/routes.py

```diff
@@ -1,2 +1,293 @@
-from flask import Blueprint
+from flask import request,Blueprint,make_response,jsonify
+from werkzeug.datastructures import Authorization
 
+from app import db
+from app.models.task import Task
+from datetime import datetime
+import requests
+import os
+
+
+
+task_bp = Blueprint("task", __name__, url_prefix="/tasks")
+
+@task_bp.route("", methods=["GET"])
+def get_tasks():
+   
+    title_query = request.args.get("title")
+    if title_query:
+        tasks = Task.query.filter_by(title=title_query)
+    
+    else:
+        tasks = Task.query.all()
+    
+    tasks_response = []
+    for task in tasks: 
+        tasks_response.append(task.to_json()) 
+    
+    if "asc" in request.full_path:   
+        sort_tasks=sorted(tasks_response, key = lambda i: i['title'])  
+        return jsonify(sort_tasks)
+    
+    elif "desc" in request.full_path:
+        sort_tasks=sorted(tasks_response, key = lambda i: i['title'], reverse=True)
+        return jsonify(sort_tasks)
+    
+    else:
+        return jsonify(tasks_response)
+
+@task_bp.route("", methods=["POST"])
+def post_tasks():
+
+    request_body = request.get_json()
+    
+    if all(keys in request_body for keys in ("title","description","completed_at"))== False:
+        return {
+    "details": "Invalid data"
+                },400
+    else:
+        new_task = Task(title=request_body["title"],
+                            description=request_body["description"],
+                            completed_at=request_body["completed_at"])
+        
+        
+
+        db.session.add(new_task) 
+        db.session.commit() 
+        
+        task = Task.query.get(new_task.task_id)
+        return  {
+            "task": task.to_json()
+        }, 201
+
+
+@task_bp.route("/<task_id>", methods=["GET"])
+def get_task(task_id):
+    task = Task.query.get(task_id)
+    
+    if task is None:
+        return make_response ("",404)
+    else:
+        return {"task":task.to_json()},200
```

Comment:
You can refine this function into a fewer lines by using Flask's own `get_or_404()` method
```suggestion
    task = Task.query.get_or_404(task_id)
    
   
    return {"task":task.to_json()},200
```